### PR TITLE
cmd: modify `gen_node_key` to print key to STDOUT

### LINKED ISSUE #5770: tendermint gen_validator and gen_node_key inconsistency
gen_validator - generates validator and prints it to STDOUT
gen_node_key - generates node key and writes it to `config/node_key.json`

Given these commands share the same prefix, I think both must have the same behavior.

### PR DESCRIPTION
**Pull request by melekes**
closes: #5770 
closes: #5769

also, include node ID in the output (#5769) and modify `NodeKey` to use
value semantics (it makes perfect sense for `NodeKey` to not be a
pointer).

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -38,6 +39,7 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 - [abci] \#5706 Added `AbciVersion` to `RequestInfo` allowing applications to check ABCI version when connecting to Tendermint. (@marbar3778)
 - [blockchain/v1] \#5728 Remove in favor of v2 (@melekes)
 - [blockchain/v0] \#5741 Relax termination conditions and increase sync timeout (@melekes)
+- [cli] \#5772 `gen_node_key` output now contains node ID (`id` field) (@melekes)
```
Comment (marbar3778) [inline CHANGELOG_PENDING.md:42]: Probably add a minor note that genNodeKey will now generate a key even if one exists.

Comment (melekes) [reply CHANGELOG_PENDING.md:42]: There's a note in the breaking section

**Review by marbar3778 — APPROVED** [review]
utACK.

**Review by cmwaters — APPROVED** [review]
LGTM

File: cmd/tendermint/commands/gen_node_key.go [hunk 2]
```diff
@@ -5,28 +5,28 @@ import (
...
-	nodeKey, err := p2p.LoadOrGenNodeKey(nodeKeyFile)
+	bz, err := tmjson.Marshal(nodeKey)
 	if err != nil {
-		return err
+		return fmt.Errorf("nodeKey -> json: %w", err)
 	}
-	fmt.Println(nodeKey.ID())
+
+	fmt.Printf(`%v
+`, string(bz))
```
Comment (alexanderbez) [inline cmd/tendermint/commands/gen_node_key.go:29]:
```suggestion
	fmt.Printf("%s\n", string(bz))
```

**Review by alexanderbez — APPROVED** [review]
🎉

File: cmd/tendermint/commands/gen_node_key.go (hunk 2 above)
Comment (alexanderbez) [reply cmd/tendermint/commands/gen_node_key.go:29]: nit: we can/should do this for the rest?

**Pull request merged**